code/modules/projectiles/guns/energy/pulse.dm

```diff
@@ -10,6 +10,7 @@
 	charge_cost=100
 	max_shots = 20 // This is cut in half by "DESTROY" mode.
 	sel_mode = 2
+	w_class = 4

```

Comment:
Can the pulse rifle still melt walls? If so, I'd go so far as to make this w_class = 5 but that is just my opinion. You only get half the shots with the weapon set to "DESTROY" anyway so this doesn't matter a lot.